src/server/rpc/procedures/NPlayer/NPlayer.js

```diff
@@ -0,0 +1,151 @@
+// This is the NPlayer game RPC. It ensures round-robin turn taking with N players.
+
+'use strict';
+
+var R = require('ramda'),
+    debug = require('debug'),
+    log = debug('NetsBlox:RPCManager:NPlayer:log'),
+    trace = debug('NetsBlox:RPCManager:NPlayer:trace'),
+    Constants = require('../../../../common/Constants'),
+    info = debug('NetsBlox:RPCManager:NPlayer:info');
+
+/**
+ * NPlayerRPC - This constructor is called on the first request to an RPC
+ * from a given room
+ *
+ * @constructor
+ * @return {undefined}
+ */
+var NPlayerRPC = function() {
+    this.active = null;
+    this.previous = null;
+    this.players = [];
+};
+
+/**
+ * Return the path to the given RPC
+ *
+ * @return {String}
+ */
+NPlayerRPC.getPath = function() {
+    return '/NPlayer';
+};
+
+/**
+ * This function is used to expose the public API for RPC calls
+ *
+ * @return {Array<String>}
+ */
+NPlayerRPC.getActions = function() {
+    return [
+            'start',  // start (or restart) the game
+            'getN',  // Get N
+            'getActive',  // get the active Player's role id
+            'getNext',  // get the next Player's role id
+            'getPrevious',  // get the previous Player's role id
+            'endTurn']  // Signal end of turn
+};
+
+// Actions
+
+// start the game
+NPlayerRPC.prototype.start = function(req, res) {
+
+	// populate the players list
+    this.players = [];
+    req.netsbloxSocket._room.sockets()
```

Comment:
This works but might be more clear with just a `map` call:
```
req.netsbloxSocket._room.sockets().map(socket => {
    return {
        role: socket.roleId,
        socket: socket
    };
});
```
No need to change it if you don't want to though.